components/ScriptLibrary/include/LHVMTypes.h

```diff
@@ -0,0 +1,371 @@
+/******************************************************************************
+ * Copyright (c) 2018-2024 openblack developers
+ *
+ * For a complete list of all authors, please refer to contributors.md
+ * Interested in contributing? Visit https://github.com/openblack/openblack
+ *
+ * openblack is licensed under the GNU General Public License version 3.
+ *******************************************************************************/
+
+#pragma once
+
+#include <cassert>
+#include <cstdint>
+
+#include <array>
+#include <functional>
+#include <map>
+#include <string>
+
+namespace openblack::LHVM
+{
+
+static const std::array<std::string, 31> k_OpcodeNames = {
+    "END",    "JZ",   "PUSH", "POP",       "ADD",       "SYS",        "SUB",       "NEG",  "MUL", "DIV", "MOD",
+    "NOT",    "AND",  "OR",   "EQ",        "NE",        "GE",         "LE",        "GT",   "LT",  "JMP", "SLEEP",
+    "EXCEPT", "CAST", "RUN",  "ENDEXCEPT", "RETEXCEPT", "FAILEXCEPT", "BRKEXCEPT", "SWAP", "LINE"};
+
+enum class LHVMVersion : uint32_t
+{
+	BlackAndWhite = 7,
+	CreatureIsle = 8,
+	BlackAndWhite2 = 12,
+
+	Last = 0xFFFFFFFF
+};
+
+enum class DataType : uint32_t
+{
+	NONE = 0,
+	INT,
+	FLOAT,
+	VECTOR,
+	OBJECT,
+	UNK2,
+	BOOLEAN,
+	UNK4,
+
+	Last = 0xFFFFFFFF
+};
+static_assert(sizeof(DataType) == 4);
+
+static const std::array<std::string, 8> k_DataTypeNames = {"NONE",   "INT",  "FLOAT",   "VECTOR",
+                                                           "OBJECT", "UNK2", "BOOLEAN", "UNK4"};
+
+static const std::array<std::string, 8> k_DataTypeChars = {"", "I", "F", "V", "O", "", "B", ""};
+
+class VMValue
+{
+public:
+	VMValue()
+	    : floatVal(0.0f)
+	{
+	}
+
+	explicit VMValue(float value)
+	    : floatVal(value)
+	{
+	}
+
+	explicit VMValue(int32_t value)
+	    : intVal(value)
+	{
+	}
+
+	explicit VMValue(uint32_t value)
+	    : uintVal(value)
+	{
+	}
+
+	union
+	{
+		float floatVal;
+		int32_t intVal;
+		uint32_t uintVal;
+	};
+};
+static_assert(sizeof(VMValue) == 4);
+
+class VMVar
+{
+public:
+	VMVar(DataType type, VMValue value, std::string name)
+	    : type(type)
+	    , value(value)
+	    , name(std::move(name))
+	{
+	}
+
+	DataType type;
+	union
+	{
+		VMValue value;
+		float floatVal;
+		int32_t intVal;
+		uint32_t uintVal;
+	};
+	std::string name;
+};
+
+class VMStack
+{
+public:
+	VMStack() = default;
+
+	uint32_t count {0};
+	std::array<VMValue, 32> values {};
+	std::array<DataType, 32> types {};
+	uint32_t pushCount {0};
+	uint32_t popCount {0};
+};
+static_assert(sizeof(VMStack) == 268);
+
+enum class ScriptType : uint32_t
+{
+	SCRIPT = 1,
+	HELP = 2,
+	CHALLENGE_HELP = 4,
+	TEMPLE_HELP = 8,
+	TEMPLE_SPECIAL = 16,
+	MULTIPLAYER_HELP = 64,
+
+	ALL = 0xFFFFFFFF
```

Comment:
Many of your enums are in C style ALLCAPS, is this a requirement for the lhvm language or can it be changed to CamelCase, like we have for all our other enums (Barring some exceptions)?

Reply:
I can rename them without side effects.